### ISSUE TITLE
Wrong prediction of Yolov11s Object Detection model after converting .pt to onnx to TensorRT

### ISSUE THREAD

**Issue by AbhisekRao** [ultralytics/ultralytics#19968]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

While converting the Yolov11s Object Detection model from .pt to Onnx using the below code, I am getting below warning while conversion but still the model is getting converted. While using the model with Deepstream the prediction are incorrect and also the accuracy is low. 

I am doing the conversion and quantization on Nvidia Jetson AGX Orion 32GB device which has deepstream 7.0. 

Can someone help me to point out where i am doing wrong and how to fix the same.

**Error** 
[04/01/2025-21:40:21] [W] [TRT] onnx2trt_utils.cpp:372: Your ONNX model has been generated with INT64 weights, while TensorRT does not natively support INT64. Attempting to cast down to INT32.
[04/01/2025-21:40:21] [W] [TRT] onnx2trt_utils.cpp:400: One or more weights outside the range of INT32 was clamped
[04/01/2025-21:40:21] [W] [TRT] Tensor DataType is determined at build time for tensors not marked as input or output.
[04/01/2025-21:40:21] [I] Finished parsing network model. Parse time: 0.113083

**Command to convert the ONNX model to TensorRT**
trtexec --onnx=model_v1.pt.onnx --saveEngine=model-gsteel-scrap-fp16.engine --fp16 --explicitBatch --minShapes=input:1x3x640x640 --optShapes=input:4x3x640x640 --maxShapes=input:8x3x640x640

**Python Code to convert .pt to ONNX**

_Github URL from where the below code was used_

[https://github.com/marcoslucianops/DeepStream-Yolo/blob/master/utils/export_yoloV8.py](url)

`import os
import sys
import onnx
import torch
import torch.nn as nn
from copy import deepcopy

import ultralytics.utils # type: ignore
import ultralytics.models.yolo # type: ignore
import ultralytics.utils.tal as _m # type: ignore

sys.modules['ultralytics.yolo'] = ultralytics.models.yolo
sys.modules['ultralytics.yolo.utils'] = ultralytics.utils

def _dist2bbox(distance, anchor_points, xywh=False, dim=-1):
    lt, rb = distance.chunk(2, dim)
    x1y1 = anchor_points - lt
    x2y2 = anchor_points + rb
    return torch.cat((x1y1, x2y2), dim)

_m.dist2bbox.__code__ = _dist2bbox.__code__

class DeepStreamOutput(nn.Module):
    def __init__(self):
        super().__init__()

    def forward(self, x):
        x = x.transpose(1, 2)
        boxes = x[:, :, :4]
        scores, labels = torch.max(x[:, :, 4:], dim=-1, keepdim=True)
        return torch.cat([boxes, scores, labels.to(boxes.dtype)], dim=-1)

def yolov8_export(weights, device, inplace=True, fuse=True):
    ckpt = torch.load(weights, map_location='cpu')
    ckpt = (ckpt.get('ema') or ckpt['model']).to(device).float()
    if not hasattr(ckpt, 'stride'):
        ckpt.stride = torch.tensor([32.])
    if hasattr(ckpt, 'names') and isinstance(ckpt.names, (list, tuple)):
        ckpt.names = dict(enumerate(ckpt.names))
    model = ckpt.fuse().eval() if fuse and hasattr(ckpt, 'fuse') else ckpt.eval()
    for m in model.modules():
        t = type(m)
        if hasattr(m, 'inplace'):
            m.inplace = inplace
        elif t.__name__ == 'Upsample' and not hasattr(m, 'recompute_scale_factor'):
            m.recompute_scale_factor = None
    model = deepcopy(model).to(device)
    for p in model.parameters():
        p.requires_grad = False
    model.eval()
    model.float()
    model = model.fuse()
    for k, m in model.named_modules():
        if m.__class__.__name__ in ('Detect', 'RTDETRDecoder'):
            m.dynamic = False
            m.export = True
            m.format = 'onnx'
        elif m.__class__.__name__ == 'C2f':
            m.forward = m.forward_split
    return model

def suppress_warnings():
    import warnings
    warnings.filterwarnings('ignore', category=torch.jit.TracerWarning)
    warnings.filterwarnings('ignore', category=UserWarning)
    warnings.filterwarnings('ignore', category=DeprecationWarning)
    warnings.filterwarnings('ignore', category=FutureWarning)
    warnings.filterwarnings('ignore', category=ResourceWarning)

def main(args):
    suppress_warnings()

    print(f'\nStarting: {args.weights}')

    print('Opening YOLOv8 model')

    device = torch.device('cpu')
    model = yolov8_export(args.weights, device)

    if len(model.names.keys()) > 0:
        print('Creating labels.txt file')
        with open('labels.txt', 'w', encoding='utf-8') as f:
            for name in model.names.values():
                f.write(f'{name}\n')

    model = nn.Sequential(model, DeepStreamOutput())

    img_size = args.size * 2 if len(args.size) == 1 else args.size

    onnx_input_im = torch.zeros(args.batch, 3, *img_size).to(device)
    onnx_output_file = f'{args.weights}.onnx'

    dynamic_axes = {
        'input': {
            0: 'batch'
        },
        'output': {
            0: 'batch'
        }
    }

    print('Exporting the model to ONNX')
    torch.onnx.export(
        model, onnx_input_im, onnx_output_file, verbose=False, opset_version=args.opset, do_constant_folding=True,
        input_names=['input'], output_names=['output'], dynamic_axes=dynamic_axes if args.dynamic else None
    )

    if args.simplify:
        print('Simplifying the ONNX model')
        import onnxslim
        model_onnx = onnx.load(onnx_output_file)
        model_onnx = onnxslim.slim(model_onnx)
        onnx.save(model_onnx, onnx_output_file)

    print(f'Done: {onnx_output_file}\n')

def parse_args():
    import argparse
    parser = argparse.ArgumentParser(description='DeepStream YOLOv8 conversion')
    parser.add_argument('-w', '--weights', required=True, help='Input weights (.pt) file path (required)')
    parser.add_argument('-s', '--size', nargs='+', type=int, default=[640], help='Inference size [H,W] (default [640])')
    parser.add_argument('--opset', type=int, default=17, help='ONNX opset version')
    parser.add_argument('--simplify', action='store_true', help='ONNX simplify model')
    parser.add_argument('--dynamic', action='store_true', help='Dynamic batch-size')
    parser.add_argument('--batch', type=int, default=1, help='Static batch-size')
    args = parser.parse_args()
    if not os.path.isfile(args.weights):
        raise SystemExit('Invalid weights file')
    if args.dynamic and args.batch > 1:
        raise SystemExit('Cannot set dynamic batch-size and static batch-size at same time')
    return args

if __name__ == '__main__':
    args = parse_args()
    main(args)`

**Comment by UltralyticsAssistant** [comment]
👋 Hello @AbhisekRao, thank you for sharing your detailed issue and for using Ultralytics 🚀! We appreciate your effort in providing the context and code.

Based on your description, it seems like you're encountering challenges with predictions after converting a YOLO11s model from `.pt` to ONNX and then to TensorRT. Here's how we can assist:

- If this is a 🐛 Bug Report, please provide a [minimum reproducible example (MRE)](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug the issue. This would include a small code snippet, example inputs, and any necessary files to replicate the problem.

- If this is a ❓ Question about custom usage or configurations, please provide additional details such as:
  - Parameters used during training (e.g., dataset size, epochs, augmentation settings, etc.).
  - Example outputs showing the incorrect predictions or reduced accuracy.
  - Any modifications to the original YOLO11s model or pipeline.

In the meantime, here are some steps to ensure you're working with the latest tools and configurations:

## Upgrade

Ensure you have the latest `ultralytics` package and dependencies installed in a [**Python>=3.8**](https://www.python.org/) environment:

```bash
pip install -U ultralytics
```

## Environments

YOLO models can be run in various environments. For testing, you might consider one of these options with pre-configured dependencies:

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Resources

We recommend checking out the following resources for additional guidance:
- [Docs](https://docs.ultralytics.com/): Contains detailed explanations and examples for usage and export.
- [Model Export Guide](https://docs.ultralytics.com/modes/export/): Step-by-step instructions for exporting YOLO models to ONNX and TensorRT formats.
- [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/): Ensure your training process aligns with recommended practices.

## Community Support

Join the Ultralytics community for more insights and help:
- [Discord](https://discord.com/invite/ultralytics) 🎧 for real-time discussions.
- [Discourse](https://community.ultralytics.com/) for in-depth topics.
- [Subreddit](https://reddit.com/r/Ultralytics) for knowledge sharing with fellow users.

## Status

Check the status of our Continuous Integration (CI) tests to ensure core functionalities are working:

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

An Ultralytics engineer will review your issue soon and provide further assistance. Thank you for your patience! 😊
